### PR TITLE
OPENEUROPA-1182: Theme "Department" RDF entity

### PR DESCRIPTION
**Pull request by upchuk**

### CODE REVIEW

File: modules/europa_demo_core/src/Plugin/PageHeaderMetadata/DepartmentPageHeaderMetadata.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ * )
+ */
+class DepartmentPageHeaderMetadata extends EntityCanonicalRoutePageHeaderMetadata {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function applies(): bool {
+    $entity = $this->getEntityFromCurrentRoute();
+    return !empty($entity) && $entity->getEntityTypeId() == 'rdf_entity' && $entity->bundle() == 'oe_department';
```
Comment (drupol) [inline modules/europa_demo_core/src/Plugin/PageHeaderMetadata/DepartmentPageHeaderMetadata.php]: Shouldn't we use `===` and `[] !== $entity` ?

File: modules/europa_demo_core/src/Plugin/PageHeaderMetadata/DepartmentPageHeaderMetadata.php [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+    $metas = [];
+
+    /** @var \Drupal\taxonomy\TermInterface $department_type */
+    $department_type = $entity->get('oe_department_type')->entity;
+    $metas[] = $department_type->label();
+
+    /** @var \Drupal\taxonomy\TermInterface[] $subjects */
+    $subjects = $entity->get('oe_department_subject')->referencedEntities();
+    $subject_metas = [];
```
Comment (drupol) [inline modules/europa_demo_core/src/Plugin/PageHeaderMetadata/DepartmentPageHeaderMetadata.php]: For the beauty of it, I would have used array_map() here.
```php
$subject_metas = array_map(
  function ($item) {return $item->label();},
  $subjects
);
```

File: modules/europa_demo_core/src/Plugin/PageHeaderMetadata/DepartmentPageHeaderMetadata.php [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    foreach ($subjects as $subject) {
+      $subject_metas[] = $subject->label();
+    }
+    if ($subject_metas) {
```
Comment (drupol) [inline modules/europa_demo_core/src/Plugin/PageHeaderMetadata/DepartmentPageHeaderMetadata.php]: I don't think this condition is necessary, `$subject_metas` will always be defined.
Even if `$subjects_metas` is empty, it won't change anything, so we can get rid of it.

See: https://3v4l.org/mELAZ

File: modules/europa_demo_core/src/Plugin/PageHeaderMetadata/DepartmentPageHeaderMetadata.php (hunk 2 above)
Comment (drupol) [reply modules/europa_demo_core/src/Plugin/PageHeaderMetadata/DepartmentPageHeaderMetadata.php]: I would have even used a better variable like $subject_labels

Comment (upchuk) [reply modules/europa_demo_core/src/Plugin/PageHeaderMetadata/DepartmentPageHeaderMetadata.php]: I don't see how array_map() is more beautiful. I will change the var name though, you are right.

Comment (drupol) [reply modules/europa_demo_core/src/Plugin/PageHeaderMetadata/DepartmentPageHeaderMetadata.php]: True, it's a matter of taste. Using array_map() is so much easier to read for me that I'm trying to encourage people to use it :-)

**Pull request merged**